src/sentry/web/frontend/oauth_token.py

```diff
@@ -42,28 +45,46 @@ def error(self, request: HttpRequest, name, reason=None, status=400):
 
     @never_cache
     def post(self, request: HttpRequest) -> HttpResponse:
+        client_details = self._get_client_details_if_valid_request(request=request)
+
+        if not isinstance(client_details, dict):
+            return client_details  # client_details contains an error response.
+
+        if client_details["grant_type"] == GrantTypes.AUTHORIZATION:
+            return self._get_access_tokens(
```

Comment:
Same comment with these helpers

Lets have the helpers pull the relevant data we need - but have this `post` be the one to return the real response

Will make it easier to test also. We can simply test for the values in the dicts returned vs. the response object

Reply:
Is there a problem with the current way though? I don't see how it would be better this way, testing is already quite straightforward

Reply:
It might be _subjectively_ better 🤷 

My thought - if we have all the helpers return a uniform dict, then we don't need to worry about which ones are returning http responses or not.
Also - if we're collecting the helper results, then explicitly returning http responses, we never need to worry about passing the helper responses straight through to the return, which makes it very clear/simple what this method is returning (rather than chasing down through a bunch of helpers to determine what the response is)